src/main/java/com/shootoff/camera/cameratypes/PS3EyeCamera.java

```diff
@@ -0,0 +1,621 @@
+//written by ifly53e
+
+package com.shootoff.camera.cameratypes;
+
+import com.sun.jna.Library;
+
+import com.sun.jna.Native;
+import com.sun.jna.Pointer;
+import com.sun.jna.PointerType;
+
+import javafx.application.Platform;
+import javafx.beans.value.ChangeListener;
+import javafx.beans.value.ObservableValue;
+import javafx.geometry.Insets;
+import javafx.scene.Group;
+import javafx.scene.Scene;
+import javafx.scene.control.CheckBox;
+import javafx.scene.control.Label;
+import javafx.scene.control.Slider;
+import javafx.scene.layout.GridPane;
+import javafx.scene.paint.Color;
+import javafx.stage.Stage;
+
+import java.awt.Dimension;
+import java.awt.image.BufferedImage;
+
+import java.util.Optional;
+
+import org.opencv.core.CvType;
+import org.opencv.core.Mat;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.shootoff.camera.CameraFactory;
+import com.shootoff.camera.CameraManager;
+import com.shootoff.camera.CameraView;
+import com.shootoff.camera.shotdetection.JavaShotDetector;
+import com.shootoff.camera.shotdetection.NativeShotDetector;
+import com.shootoff.camera.shotdetection.OptiTrackShotDetector;
+import com.shootoff.camera.shotdetection.ShotDetector;
+import com.shootoff.config.Configuration;
+
+public class PS3EyeCamera implements Camera {
+	private static final Logger logger = LoggerFactory.getLogger(PS3EyeCamera.class);
+
+	private static boolean initialized = false;
+	protected CameraState cameraState;
+	protected Optional<CameraEventListener> cameraEventListener = Optional.empty();
+	protected long currentFrameTimestamp = -1;
+	private Dimension dimension = null;
+	private int viewWidth = 0;
+	private int viewHeight = 0;
+
+	static int size_x = 640;
+	static int size_y = 480;
+
+	static eyecam.ps3eye_t ps3ID = null;
+	static boolean opened = false;
+	static int frameCount = 0;
+	static boolean closed = true;
+	static double ps3FPS = 30;
+	private long lastCameraTimestamp = -1;
+	private long lastFrameCount = 0;
+
+	static eyecam eyecamLib = (eyecam) Native.loadLibrary("eyecam", eyecam.class);
```

Comment:
My understanding is that this will throw an exception if the dll is not present, which will not work on Linux.  This should be a dynamic call that exits if the module can't be loaded.  See OptiTrackCamera for an example of failing safe on "UnsatisfiedLinkError".